### PR TITLE
ci: bump bitcoind version in functional tests to 26.0

### PR DESCRIPTION
**Pull request by darosior**
I thought there was some hacks i do in the functional tests i needed to update but apparently i was mistaken.

### CODE REVIEW

**Comment by darosior** [comment]
Ran the whole test suite a dozen times locally without encountering any flaky test.

File: .cirrus.yml [hunk 1]
```diff
@@ -54,10 +54,10 @@ task:
       export IS_BITCOIND_25=0
```
Comment (jp1ac4) [inline .cirrus.yml:54]: Should we rename this variable to something like `IS_BITCOIND_LATEST`?

Comment (darosior) [reply .cirrus.yml]: It's not the latest we need but anything above 24. Renamed.

**Comment by darosior** [comment]
self-ACK bdf8b8f625030697ca58680fd0c99c3dd3bf77af - ran ci one last time and didn't notice any flakiness

**Pull request merged**